pkg/hnsw/hnsw.go

```diff
@@ -59,3 +59,54 @@ func (h *Hnsw) EntryTopLayer() int {
 func (h *Hnsw) SpawnLayer() int {
 	return int(math.Floor(-math.Log(rand.Float64() * h.levelMultiplier)))
 }
+
+// w must be a max euc queue
+func (h *Hnsw) searchLayer(q Vector, ef, layerId int, w *EucQueue) {
```

Comment:
return/handle the `err`s

Reply:
The only error raised was when computing the `Euclid Distance`. However, once merging the latest changes which refactors the euclid distance to only return the distance, we don't have any errors to propagate in `SearchLayer`.